src/Persistence/SQL.php

```diff
@@ -441,7 +440,8 @@ public function initQueryConditions(Model $m, Query $q): Query
                 }
                 $q->where($cond[0], $cond[1]);
             } else {
-                if ($cond[0] instanceof Field) {
+                // "like" condition do not need typecasting to field type!
+                if ($cond[0] instanceof Field && strtolower($cond[1]) != 'like') {
```

Comment:
Use `in_array` and fix for regexp as well.

Reply:
ok, fixed